src/lib/testBase.ts

```
@@ -0,0 +1,50 @@
+import { IExecutable, ProviderInterface } from "../@types"
+
+interface State {
+    hasRan: boolean
+}
+
+export abstract class TestBase implements IExecutable {
+    public state: State = { hasRan: false }
+    protected beaconData: unknown
+    constructor(protected provider: ProviderInterface, protected config: unknown) {}
+
+    /**
+     * This is the logic function for conducting an individual test.
+     */
+    execute(): Promise<unknown> {
+        const result = this.makeTestSteps()
+            .then((data): unknown => {
+                const result: any = this.provider.makeBeaconData(this.config, data)
+                this.setBeaconData(result)
+                return result
+            })
+            .then((data): void => this.provider.sendBeacon(this.config, this.encodeBeaconData(data)))
+            .then((): unknown => this.beaconData)
+            .catch((e): Promise<unknown> => {
+                console.log(`Error caught in TestBase.execute: ${e}`)
+                console.log(result)
+                return Promise.resolve<unknown>('Replace me!')
+            })
```

In insighst.js we intentionally swallow any errors to be a good third-party citizen and not let let bubble up to the host page. However similar to the debug logging story, I've always wanted to to make it configurable that errors could get reported back to a provider endpoint. Something to discuss/consider.

---

I've removed the console logging from this handler, but it still doesn't do anything useful besides mark the Test object as failed. I'll leave this for future work.